tools/server/webui/src/utils/llama-vscode.ts

```diff
@@ -33,12 +44,18 @@ export const useVSCodeContext = (textarea: ChatTextareaApi) => {
           });
         }
         textarea.focus();
+        if (onSend && data?.text) {
+          // Use setTimeout to ensure state updates are processed
+          setTimeout(() => {
+            onSend();
+          }, 50);
+        }
```

Comment:
Tbh I don't quite like this approach because it makes the logic looks kinda circular dependency.

A cleaner approach is to simply extend `useChatTextarea.ref` to temporary hold a callback let's say `textarea.onSubmit`, then call it here:

```suggestion
        textarea.focus();
        textarea.onSubmit();
```

Reply:
I agree. It is cleaner.